exchanges/btcmarkets/btcmarkets.go

```diff
@@ -52,316 +60,465 @@ type BTCMarkets struct {
+}
 
-	err := b.SendAuthenticatedRequest(http.MethodGet, btcMarketsAccountBalance, nil, &balance)
-	if err != nil {
-		return nil, err
-	}
+// RemoveOrder removes a given order
+func (b *BTCMarkets) RemoveOrder(id string) (CancelOrderResp, error) {
+	var resp CancelOrderResp
+	return resp, b.SendAuthenticatedRequest(http.MethodDelete, btcMarketsOrders+"/"+id,
+		nil, &resp)
+}
 
-	// All values are returned in Satoshis, even for fiat currencies.
-	for i := range balance {
-		balance[i].Balance /= common.SatoshisPerBTC
-		balance[i].PendingFunds /= common.SatoshisPerBTC
-	}
-	return balance, nil
+// ListWithdrawals lists the withdrawal history
+func (b *BTCMarkets) ListWithdrawals() ([]TransferData, error) {
+	var resp []TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsWithdrawals, nil, &resp)
+}
+
+// GetWithdrawal gets withdrawawl info for a given id
+func (b *BTCMarkets) GetWithdrawal(id string) (TransferData, error) {
+	var resp TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsWithdrawals+"/"+id,
+		nil, &resp)
 }
 
-// GetTradingFee returns the account's trading fee for a currency pair
-func (b *BTCMarkets) GetTradingFee(base, quote currency.Code) (TradingFee, error) {
-	var tradingFee TradingFee
-	path := fmt.Sprintf(btcMarketsTradingFee, base, quote)
-	return tradingFee, b.SendAuthenticatedRequest(http.MethodGet, path, nil, &tradingFee)
+// ListDeposits lists the deposit history
+func (b *BTCMarkets) ListDeposits() ([]TransferData, error) {
+	var resp []TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsDeposits, nil, &resp)
 }
 
-// WithdrawCrypto withdraws cryptocurrency into a designated address
-func (b *BTCMarkets) WithdrawCrypto(amount float64, currency, address string) (string, error) {
-	newAmount := int64(amount * float64(common.SatoshisPerBTC))
+// GetDeposit gets deposit info for a given ID
+func (b *BTCMarkets) GetDeposit(id string) (TransferData, error) {
+	var resp TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsDeposits+"/"+id,
+		nil, &resp)
+}
 
-	req := WithdrawRequestCrypto{
-		Amount:   newAmount,
-		Currency: strings.ToUpper(currency),
-		Address:  address,
-	}
+// ListTransfers lists the past asset transfers
+func (b *BTCMarkets) ListTransfers() ([]TransferData, error) {
+	var resp []TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsTransfers, nil, &resp)
+}
+
+// GetTransfer gets asset transfer info for a given ID
+func (b *BTCMarkets) GetTransfer(id string) (TransferData, error) {
+	var resp TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsTransfers+"/"+id,
+		nil, &resp)
+}
 
-	resp := Response{}
-	err := b.SendAuthenticatedRequest(http.MethodPost,
-		btcMarketsWithdrawCrypto,
+// FetchDepositAddress gets deposit address for the given asset
+func (b *BTCMarkets) FetchDepositAddress(assetName string) (DepositAddress, error) {
```

Comment:
Another Method that will not work with due to it being a get request with query string

```
BTC Markets exchange request path: https://api.btcmarkets.net/v3/addresses requires rate limiter: true
BTC Markets exchange request header [Accept-Charset]: [UTF-8]
BTC Markets exchange request header [Content-Type]: [application/json]
BTC Markets exchange request header [Bm-Auth-Apikey]: [xxx]
BTC Markets exchange request header [Bm-Auth-Timestamp]: [1574032233192]
BTC Markets exchange request header [Bm-Auth-Signature]: [xxx]
BTC Markets exchange request header [Accept]: [application/json]
BTC Markets exchange request type: GET
BTC Markets exchange request body: {"assetName":"LTC"}
BTC Markets request. Job complete.
--- FAIL: TestFetchDepositAddress (0.12s)
    btcmarkets_test.go:300: unsuccessful HTTP status code: 400
        BTC Markets exchange raw response: {"message":"Bad Request"}
    btcmarkets_test.go:302: {Address: AssetName:}